toolkit/tools/pkg/imagecreatorlib/imagecreator_test.go

```diff
@@ -0,0 +1,262 @@
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	fileType, err := testutils.GetImageFileType(outputImageFilePath)
+	assert.NoError(t, err)
+	assert.Equal(t, "raw", fileType)
+
+	imageInfo, err := imagecustomizerlib.GetImageFileInfo(outputImageFilePath)
+	assert.NoError(t, err)
+	assert.Equal(t, "raw", imageInfo.Format)
+	assert.Equal(t, int64(1*diskutils.GiB), imageInfo.VirtualSize)
+
+	// Customize image to vhd.
+	err = imagecustomizerlib.CustomizeImageWithConfigFile(buildDir, noChangeConfigFile, outputImageFilePath, rpmSources, vhdFixedImageFilePath,
+		"vhd", false /*useBaseImageRpmRepos*/, "" /*packageSnapshotTime*/)
+	if !assert.NoError(t, err) {
+		return
+	}
+	fileType, err = testutils.GetImageFileType(vhdFixedImageFilePath)
+	assert.NoError(t, err)
+	assert.Equal(t, "vhd", fileType)
+
+	imageInfo, err = imagecustomizerlib.GetImageFileInfo(vhdFixedImageFilePath)
+	assert.NoError(t, err)
+	assert.Equal(t, "vpc", imageInfo.Format)
+	assert.Equal(t, int64(1*diskutils.GiB), imageInfo.VirtualSize)
+
+	// Customize image to vhdx.
+	err = imagecustomizerlib.CustomizeImageWithConfigFile(buildDir, noChangeConfigFile, vhdFixedImageFilePath, rpmSources, vhdxImageFilePath,
+		"vhdx", false /*useBaseImageRpmRepos*/, "" /*packageSnapshotTime*/)
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	fileType, err = testutils.GetImageFileType(vhdxImageFilePath)
+	assert.NoError(t, err)
+	assert.Equal(t, "vhdx", fileType)
+
+	imageInfo, err = imagecustomizerlib.GetImageFileInfo(vhdxImageFilePath)
+	assert.NoError(t, err)
+	assert.Equal(t, "vhdx", imageInfo.Format)
+	assert.Equal(t, int64(1*diskutils.GiB), imageInfo.VirtualSize)
+}
+
+func TestCreateImageRawNoTar(t *testing.T) {
+	testTmpDir := filepath.Join(tmpDir, "TestCreateImageRaw")
+	buildDir := filepath.Join(testTmpDir, "build")
+	partitionsConfigFile := filepath.Join(testDir, "minimal-os.yaml")
+	outputImageFilePath := filepath.Join(testTmpDir, "image1.raw")
+
+	// get RPM sources
+	downloadedRpmsRepoFile := testutils.GetDownloadedRpmsRepoFile(t, testutilsDir, "3.0", false, true)
+	rpmSources := []string{downloadedRpmsRepoFile}
+
+	err := CreateImageWithConfigFile(buildDir, partitionsConfigFile, rpmSources, "", outputImageFilePath, "raw")
+
+	assert.Equal(t, "tools tar file is required for image creation", err.Error())
+}
+
+func TestCreateImageEmptyConfig(t *testing.T) {
+	testTmpDir := filepath.Join(tmpDir, "TestCreateImageRaw")
+	buildDir := filepath.Join(testTmpDir, "build")
+	outputImageFilePath := filepath.Join(testTmpDir, "image1.raw")
+
+	// create an empty config file
+	emptyConfigFile := filepath.Join(testDir, "empty-config.yaml")
+
+	downloadedRpmsRepoFile := testutils.GetDownloadedRpmsRepoFile(t, testutilsDir, "3.0", false, true)
+	rpmSources := []string{downloadedRpmsRepoFile}
+
+	err := CreateImageWithConfigFile(buildDir, "", rpmSources, "", outputImageFilePath, "raw")
+	assert.Equal(t, "open : no such file or directory", err.Error())
```

Comment:
This is a pretty unhelpful error message for the user. You should look for places to add `fmt.Errorf` to add helpful debugging information to this error message.

Similar comment for the `"EOF"` error message.